PE/__init__.py

```diff
@@ -833,21 +853,175 @@ def parseDelayImports(self):
+                        # e.g., in the same section.
+                        #
+                        # so, is `abs(*ptr - ptr) < min rva`?
+                        # this should be the case for everything except:
+                        #  - REALLY large import tables, and
+                        #  - import tables split across more than one large section
+                        #
+                        # dereference as an RVA, and if the heuristic works, vote yes.
+                        # otherwise, its probably a VA.
+                        deref = self.readPointerAtRva(ptr)
+                        if abs(deref - ptr) < self.getMinRva():
+                            # the data is nearby, so the delta is small, so its probably an RVA.
+                            votes.append(True)
+                        else:
+                            # the treating the data as an RVA resulting in a large delta,
+                            # so its probably a VA.
+                            votes.append(False)
+                    else:
+                        # for clarity: all the cases are explicit above.
+                        raise Exception("impossible")
+
+                else:
+                    # its not an RVA,
+                    # and there' no overlap between valid VA and RVA ranges,
+                    # so it must be a VA.
+                    votes.append(False)
+
+            else:
+                # for clarity: all the cases are explicit above.
+                raise Exception("impossible")
+
+            irva += isize
+
+            if not self.checkRva(irva, size=isize):
+                # if import table is at the end of the file
+                # we may run to the end.
+                break
+
+            x.vsParse(self.readAtRva(irva, isize))
+
+        logger.debug("delayed imports: votes for RVAs: %d", sum([1 for vote in votes if vote]))
+        logger.debug("delayed imports: votes for VAs: %d", sum([1 for vote in votes if not vote]))
+
+        # more votes for RVA than for VA
+        return sum([1 for vote in votes if vote]) > sum([1 for vote in votes if not vote])
+
+    def parseImportTable(self, x, irva, flavor="import table", uses_rva=True):
         '''
         Parse a standard or delayed import table, adding to imports_list.
         Start with x and irva set to the first entry in the table.
         '''
+        if flavor not in ("import table", "delay import table"):
+            raise ValueError("unexpected flavor: " + flavor)
+
         imports_list = []
         isize = len(x)
 
         while True:
-            if is_imports:
+            if flavor == "import table":
                 entry_name = x.Name
-            else:
+            elif flavor == "delay import table":
                 entry_name = x.rvaDLLName
+            else:
+                raise ValueError("unexpected flavor: " + flavor)
```

Comment:
Since you have the check for the type above, are these additional checks necessary?

Reply:
when dealing with enum-style things in python i like to explicitly name each case and raise an exception otherwise. this helps catch bugs in the future if another enum member is added but the author forgets to update a switch/case. using a single `if/else` hides the bug and goes to the wrong logic block.